q2_diversity/plugin_setup.py

```diff
@@ -240,6 +240,71 @@
     citations=[citations['legendrelegendre']]
 )
 
+plugin.methods.register_function(
+    function=q2_diversity.tsne,
+    inputs={'distance_matrix': DistanceMatrix},
+    parameters={
+        'number_of_dimensions': Int % Range(1, None),
+        'perplexity': Float % Range(1, None),
+        'early_exaggeration': Float % Range(0, None),
+        'learning_rate': Float % Range(10.0, None),
+        'n_iter': Int % Range(1, None)
+    },
+    outputs=[('tsne', PCoAResults)],
+    input_descriptions={
+        'distance_matrix': ('The distance matrix on which t-SNE should be '
+                            'computed.')
+    },
+    parameter_descriptions={
+        'number_of_dimensions': "Dimensions to reduce the distance matrix to.",
+        'perplexity':           "Provide the Balance between local and global"
+                                "structure. Low values concentrate on local"
+                                "structure. Large values sacrafice local "
+                                "details for a broader global embedding."
+                                "The default value is 25 to achive  better"
+                                "results for small microbiome datasets",
+        'early_exaggeration':   "Affects the tightnes of the shown clusters"
+                                "Larger values increase the distance between"
+                                "natural clusters in the embedded space",
+        'learning_rate':        "controls how much the weights are adjusted "
+                                "at each update",
+    },
+    output_descriptions={'tsne': 'The resulting t-SNE matrix.'},
+    name='t-distributed stochastic neighbor embedding',
+    description=("Apply t-distributed stochastic neighbor embedding."),
+
+)
+
+plugin.methods.register_function(
+    function=q2_diversity.umap,
+    inputs={'distance_matrix': DistanceMatrix},
+    parameters={
+        'number_of_dimensions': Int % Range(1, None),
```

Comment:
Same question as above - should the minimum allowed value be 2?

Reply:
Yes. You are absolutely right! Theoretically, it is possible, but I don't see any use cases where it would be beneficial to present or explore data in just one dimension.